src/_core/fetchInBatches.ts

```diff
@@ -0,0 +1,39 @@
+import invariant from 'tiny-invariant';
+
+import { chunk } from './chunk';
+
+export function fetchInBatches<T>(
+  allItems: unknown[],
+  chunkSize: number,
+  fetchFn: (items: unknown[]) => ReturnType<typeof fetch>
+): Promise<T[]> {
```

Comment:
`fetchFn` should return `Promise<T[]>` instead. This function doesn't need to know anything about `fetch`